services/wallet/database.go

```diff
@@ -273,6 +273,67 @@ SELECT blocks.hash, blk_number FROM accounts_to_blocks JOIN blocks ON blk_number
 	return nil, err
 }
 
+type Token struct {
+	Address  common.Address `json:"address"`
+	Name     string         `json:"name"`
+	Symbol   string         `json:"symbol"`
+	Decimals uint           `json:"decimals"`
+	Color    string         `json:"color"`
+}
+
+func (db *Database) GetCustomTokens() ([]*Token, error) {
+	rows, err := db.db.Query(`SELECT address, name, symbol, decimals, color FROM tokens WHERE network_id = ?`, db.network)
```

Comment:
You need to close `rows` otherwise it will hang eventually.
`defer rows.Close()`
although close might return an error, generally we ignore it, otherwise before any return

Reply:
It's important to do it AFTER `err` check otherwise `rows` might be missing.